libs-scala/oracle-testing/src/main/scala/testing/oracle/OracleAround.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.testing.oracle
+
+import com.daml.ports._
+import java.sql._;
+import scala.util.{Random, Using}
+
+private[oracle] final case class User(name: String, pwd: String)
+
+trait OracleAround {
+  Class.forName("oracle.jdbc.OracleDriver")
+
+  @volatile
+  private var systemUser: String = _
+  @volatile
+  private var systemPwd: String = _
+  @volatile
+  private var port: Port = _
+
+  def oraclePort: Port = port
+
+  protected def connectToOracle(): Unit = {
+    systemUser = sys.env("ORACLE_USERNAME")
+    systemPwd = sys.env("ORACLE_PWD")
+    port = Port(sys.env("ORACLE_PORT").toInt)
+  }
+
+  protected def createNewRandomUser(): User = {
+    // See https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements008.htm#i27570
+    // for name restrictions.
+    val u = "u" + Random.alphanumeric.take(29).mkString("")
+    createNewUser(u)
+  }
+
+  private val pwd = "hunter2"
```

Comment:
`val`s in `trait`s can lead to weird issues based on the order of inheritance, I think this can be safely put in the `createNewUser` method as IIUC it's only used there.

Reply:
til, thx 5bbc67d903a0c298d2304bb1001e6e61fd05e3d3 Also moved away the `Class.forName` stuff into `beforeAll` which seems like a more sensible place.